### DISCUSSION TITLE
container 컴포넌트에 onClick props 추가 건의

### DISCUSSION THREAD

**Discussion by herrakam** [eRentronic/eRentronic-Client#87] [아이디어]
### 현 상황
- 컨테이너 컴포넌트에서 이벤트를 받아야 할 경우, 컨테이너 내부에 이벤트를 받는 역할 만 하는 이벤트 컨테이너를 추가해줘야 함
- styled component로 이벤트 컨테이너를 생성해줘야 함
- 이벤트 컨테이너에서 내부 컨텐츠들을 위한 정렬을 또 다시 해줘야 함 

### 의견 제시
- 조건부 props로 onClick이라는 변수명으로 onClick시 실행될 함수를 받는건 어떨까?

### 의견 반영시 변화
- container의 활용성이 크게 개선됨
- onClick으로 props 명을 받는다면, styledComponent에 onClick 이벤트를 달아주는 것과 동일하게 쓰일 수 있음
```
javascript
<Container justifyContent="center" onClick={func}>
...
</Container>
```
- 불필요한 styledComponent 생성을 방지할 수 있음

**Comment by godhyeongman** [comment]
회의 통해서 컨테이너에 onClick Option 추가하는 방향으로 결정! 현재 작업물 추가할때 업데이트 하겠습니다~
